GoogleForCreators/web-stories-wp
# Code Quality: Move StoryAnimation components to root

## Summary
- Moves animation code closer together and into `/edit-story`.
- Updates `useStoryAnimationContext()` to follow `useContextSelector()` pattern.
- Fixes some PropType warnings around the animation prop types in the dashboard

## Relevant Technical Choices
Just went with all named imports with everything exported from the root `/edit-sotry/animation/index.js` so if anything needs to be imported from the animation module can just be `import { <thing> } from '/edit-story/animation'`. This makes it a little closer to a standalone package, and using all named exports allows for better lookups of source definitions in vs code.

## To-do
NA

## User-facing changes
NA

## Testing Instructions
Nothing should have changed. If there are errors they would be build errors that CI should catch. for safe measure can run the following steps if you so please:
- go to experiments. Turn on `enableAnimations` & save.
- Hover over story with animation in dashboard. should work.
- Edit story with animation and play button should work.
- Preview the same story from the editor and the animations should play in amp.
- turn off `enableAnimations` in experiments & save
- run through all major views from above steps and make sure nothing seems broken

---



Fixes #3670
